pycparser/_ast_gen.py

```diff
@@ -253,21 +283,35 @@ def visit_Constant(self, node):
         *   Modeled after Python's own AST visiting facilities
             (the ast module of Python 3.0)
     """
+    
+    _method_cache = None
+    
     def visit(self, node):
         """ Visit a node.
         """
-        method = 'visit_' + node.__class__.__name__
-        visitor = getattr(self, method, self.generic_visit)
+        
+        method_cache = self._method_cache
+        if method_cache is None:
+            method_cache = {}
+            self._method_cache = method_cache
+            
+        visitor = method_cache.get(node.__class__.__name__, None)
+        if visitor is None:
+            method = 'visit_' + node.__class__.__name__
+            visitor = getattr(self, method, self.generic_visit)
+            
+            method_cache[node.__class__.__name__] = visitor
+            
         return visitor(node)
 
     def generic_visit(self, node):
         """ Called if no explicit visitor function exists for a
             node. Implements preorder visiting of the node.
         """
-        for c_name, c in node.children():
+        for c in node:
             self.visit(c)
-
-
+            
```

Comment:
Remove trailing whitespace here and elsewhere

Reply:
Should be fixed now.